src/main/flight/pid.c

```diff
@@ -774,6 +777,46 @@ float pidGetAirmodeThrottleOffset(void)
 }
 #endif
 
+
+// ezLanding stuff
+static bool applyEzLandingLimiting = false;
+static float ezLandingLimit = PIDSUM_LIMIT; // 500
+static float maxDeflectionAbs = 1.0f;
+
+// EzLanding factors are updated only when new Rx data is received in rc.c
+// this will cause steps in PID as the limiting value changes
+FAST_CODE_NOINLINE void calcEzLandingLimit(float maxRcDeflectionAbs)
+{
+    ezLandingLimit = PIDSUM_LIMIT;
+    applyEzLandingLimiting = false;
+    if (pidRuntime.useEzLanding && !isFlipOverAfterCrashActive()) {
+        maxDeflectionAbs = fmaxf(maxRcDeflectionAbs, mixerGetRcThrottle());
+        if (maxDeflectionAbs < pidRuntime.ezLandingThreshold) {
+            applyEzLandingLimiting = true;
+            ezLandingLimit = fmaxf(pidRuntime.ezLandingLimit, PIDSUM_LIMIT * maxDeflectionAbs / pidRuntime.ezLandingThreshold);
+            // min of default limit of 15, max of 500
+        }
+    }
+    DEBUG_SET(DEBUG_EZLANDING, 0, ezLandingLimit);
+    DEBUG_SET(DEBUG_EZLANDING, 1, maxDeflectionAbs * 100);
+}
+
+static FAST_CODE_NOINLINE void disarmOnImpact(void)
+{
+    // if both sticks are within 5% of center, check acc magnitude for impacts
+    // threshold should be highe enough to avoid unwanted disarms in the air on throttle chops
+
+    // normally the acc calculation would be done only when required, but having them here lets us log the acc magnitude in normal flight
+    float accMagnitude = sqrtf(sq(acc.accADC[Z]) + sq(acc.accADC[X]) + sq(acc.accADC[Y])) * acc.dev.acc_1G_rec - 1.0f;
+    DEBUG_SET(DEBUG_EZLANDING, 4, lrintf(accMagnitude * 10));
+
+    if (isAirmodeActivated() && maxDeflectionAbs < 0.05f && accMagnitude > pidRuntime.ezLandingDisarmThreshold) {
```

Comment:
(maxDeflectionAbs is updated in other function. It makes it hard to reason about)

Reply:
@ledvinap This disarming code is now in a separate PR #13803 , and I hope it looks better there, after feedback from yourself and others.